nevergrad/optimization/optimizerlib.py

```diff
@@ -2411,3 +2411,56 @@ def recommend(self) -> p.Parameter:
 @registry.register
 class NGOpt(NGOpt10):
     pass
+
+
+class _MSR(CM):
+    """This code applies 16 copies of NGOpt10 with random weights for the different objective functions.
+
+    Variants dedicated to multiobjective optimization by multiple monoobjective optimization.
+    """
+
+    def __init__(
+        self,
+        parametrization: IntOrParameter,
+        budget: tp.Optional[int] = None,
+        num_workers: int = 1,
+        num_msr: int = 9,
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        assert budget is not None
+        self.num_optims = num_msr
+        self.optims = [
+            NGOpt(self.parametrization, budget=1 + (budget // self.num_optims), num_workers=num_workers)
+            for _ in range(self.num_optims)
+        ]
+        self.coeffs = [
+            self.parametrization.random_state.uniform(size=self.num_objectives)
+            for _ in range(self.num_optims)
+        ]
```

Comment:
num_objectives is only known at the first `tell`, so shoulnd be used here. Also, what should happen in case of monoobjective?

Reply:
Moved to tell. Should work in the single objective case I guess, and not trivial though a part becomes useless.